_posts/2023-03-26-Kafka-IaC.md

```diff
@@ -0,0 +1,249 @@
+
+Once the credentials are set run `pulumi up` command to provision the topics in confluent cloud.
+
+### Provisioning Kafka Connectors
+
+Messages from Kafka topics can be copied to and from external applications and data systems with Kafka Connect. Connectors are of two types, 
+Source connectors - The connector that takes data from a Producer and feeds them into a topic is called a Source connector.
+Sink connectors. The connector that takes data from a Topic and delivers them to a Consumer is called a Sink Connector.
+
+Let’s provision a Kafka Sink Connector that writes data from a Kafka topic to an Azure Data Lake Storage(ADLS)
+
+The mandatory configs for provisioning a Kafka Connector are
+The name of the resource
+Environment
+Cluster
+And a couple of connector-specific configs like connector class, source topics etc.
+
+The configs can contain secrets like passwords, api keys, tokens. They are by default masked by Pulumi. Those configs have to be under `configSensitive` block in config and non sensitive configs have to be under `configNonsensitive` block.
+
+```
+let connector_args: confluent.ConnectorArgs = {
+    configNonsensitive: {
+      ["connector.class"]: "AzureDataLakeGen2Sink", # The class of the Connector. List of [supported connectors](https://docs.confluent.io/cloud/current/connectors/index.html#supported-connectors)
+      ["name"]: "Connector Name",
+      ["kafka.auth.mode"]: "KAFKA_API_KEY",
+      ["topics"]: topicNames,
+      ["input.data.format"]: "JSON",
+      ["output.data.format"]: "JSON",
+      ["time.interval"]: "HOURLY",
+      ["tasks.max"]: "2",
+      ["flush.size"]: "1000",
+      ["rotate.schedule.interval.ms"]: "3600000",
+      ["rotate.interval.ms"]: "3600000",
+      ["path.format"]: "'year'=YYYY/'month'=MM/'day'=dd/'hour'=HH",
+      ["topics.dir"]: "<Directory in ADLS>",
+    },
+    configSensitive: {
+      ["kafka.api.key"]: kafka_api_key,
+      ["kafka.api.secret"]: kafka_api_secret,
+      ["azure.datalake.gen2.account.name"]: azure_data_lake_account_name,
+      ["azure.datalake.gen2.access.key"]: azure_data_lake_access_key,
+    },
+    environment: cluster_environment,
+    kafkaCluster: cluster,
+  };
+
+  new confluent.Connector("pulumi-connector", connector_args);
+```
+
+If the Confluent Cloud cluster credentials are already set up, directly go ahead and run `pulumi up` command to provision the topics in the confluent cloud. If the Confluent Cloud cluster credentials aren’t set up, follow the steps from the topic provisioning and set them up.
+
```

Comment:
Confluent has an official terraform registry for Confluent cloud but not Confluent platform. Like that, please talk a little about which IaC tool is right for which use case